lib/pos_mempool_test.go

```diff
@@ -0,0 +1,298 @@
+	rand := rand.New(rand.NewSource(seed))
+	globalParams := _testGetDefaultGlobalParams()
+	randomIndex := uint32(rand.Intn(indexRange))
+	baseRate, bucketMultiplier := globalParams.ComputeFeeTimeBucketMinimumFeeAndMultiplier()
+	feeMin, feeMax := ComputeFeeTimeBucketRangeFromIndex(randomIndex, baseRate, bucketMultiplier)
+	txnPool := _testGetRandomMempoolTxns(rand, feeMin, feeMax, timestampRange, testCases)
+
+	// Create new FeeBucket and add the txn pool
+	bucketFeeMin, bucketFeeMax := ComputeFeeTimeBucketRangeFromIndex(randomIndex, baseRate, bucketMultiplier)
+	timeBucket := NewFeeTimeBucket(bucketFeeMin, bucketFeeMax)
+	_testBucketStandardAddTest(t, txnPool, globalParams, true,
+		func(tx *MempoolTx) {
+			require.Equal(nil, timeBucket.AddTransaction(tx))
+		},
+		func() []*MempoolTx {
+			return timeBucket.GetTransactions()
+		},
+	)
+}
+
+func _testGetDefaultGlobalParams() *GlobalParamsEntry {
+	globalParams := InitialGlobalParamsEntry
+	globalParams.MinimumNetworkFeeNanosPerKB = 1000
+	globalParams.FeeBucketRateMultiplierBasisPoints = 1000
+
+	return &globalParams
+}
+
+func _testGetRandomMempoolTxns(rand *rand.Rand, feeMin uint64, feeMax uint64, timestampRange uint64, numTxns int) []*MempoolTx {
+	txnPool := []*MempoolTx{}
+	for ii := 0; ii < numTxns; ii++ {
+		txnPool = append(txnPool, &MempoolTx{
+			FeePerKB: rand.Uint64()%(feeMax-feeMin) + feeMin,
+			Added:    time.UnixMicro(int64(rand.Uint64() % timestampRange)),
+			Hash:     NewBlockHash(RandomBytes(32)),
+		})
+	}
+	return txnPool
+}
+
+func _testMapMempoolTxnToTimeBucket(txn *MempoolTx, globalParams *GlobalParamsEntry) *FeeTimeBucket {
+	baseRate, bucketMultiplier := globalParams.ComputeFeeTimeBucketMinimumFeeAndMultiplier()
+	timeBucketIndex := ComputeFeeTimeBucketIndexFromFeeNanosPerKB(txn.FeePerKB, baseRate, bucketMultiplier)
+	bucketFeeMin, bucketFeeMax := ComputeFeeTimeBucketRangeFromIndex(timeBucketIndex, baseRate, bucketMultiplier)
+	return NewFeeTimeBucket(bucketFeeMin, bucketFeeMax)
+}
+
+func _testSortMempoolTxnsByFeeTime(txnPool []*MempoolTx, globalParams *GlobalParamsEntry, timeOnly bool) []*MempoolTx {
+	if !timeOnly {
+		sort.Slice(txnPool, func(i, j int) bool {
+			timeBucketI := _testMapMempoolTxnToTimeBucket(txnPool[i], globalParams)
+			timeBucketJ := _testMapMempoolTxnToTimeBucket(txnPool[j], globalParams)
+
+			feeComparison := feeTimeBucketComparator(timeBucketI, timeBucketJ)
+			if feeComparison == 1 {
+				return false
+			} else if feeComparison == -1 {
+				return true
+			}
+
+			timeComparison := mempoolTxTimeOrderComparator(txnPool[i], txnPool[j])
+			if timeComparison == 1 {
+				return false
+			} else if timeComparison == -1 {
+				return true
+			}
+			return true
+		})
+	} else {
+		sort.Slice(txnPool, func(i, j int) bool {
+			timeComparison := mempoolTxTimeOrderComparator(txnPool[i], txnPool[j])
+			if timeComparison == 1 {
+				return false
+			} else if timeComparison == -1 {
+				return true
+			}
+			return true
+		})
+	}
+	return txnPool
```

Comment:
I think we can simplify the sorting logic here.
```suggestion
	sort.Slice(txnPool, func(i, j int) bool {
		if !timeOnly {
			timeBucketI := _testMapMempoolTxnToTimeBucket(txnPool[i], globalParams)
			timeBucketJ := _testMapMempoolTxnToTimeBucket(txnPool[j], globalParams)

			feeComparison := feeTimeBucketComparator(timeBucketI, timeBucketJ)
			if feeComparison == 1 {
				return false
			} else if feeComparison == -1 {
				return true
			}
		}
		
		timeComparison := mempoolTxTimeOrderComparator(txnPool[i], txnPool[j])
		return timeComparison != 1
	})
	return txnPool
```